geo/services/Geo_LocationService.php

```diff
@@ -3,163 +3,78 @@
-    //         "ip"=>$data['ip'],
-    //         "country_code"=>$data['country_code'],
-    //         "country_name"=>$data['country'],
-    //         "latitude"=>$data['latitude'],
-    //         "longitude"=>$data['longitude'],
-    //         "cached"=>false
-    //     );
-
-    //     return $data;
-    // }
-
-    private function anonymizeIp($ip) {
-        if (filter_var($ip, FILTER_VALIDATE_IP) !== false) {
-            $segments = explode('.', $ip);
-
-            switch (strlen($segments[3])) {
-                case 1:
-                case 2:
-                    // Set last segment to zero
-                    $segments[3] = 0;
-                    break;
-
-                case 3:
-                    // Keep the first digit and set the last two to zero
-                    $ending = substr($segments[3], 0, 1);
-                    if (substr($segments[3], 1) == '00') {
-                        // Set two random digits if IP already ended with two zeros
-                        $ending .= rand(0, 9);
-                        $ending .= rand(0, 9);
-                    } else {
-                        $ending .= '00';
-                    }
-                    $segments[3] = $ending;
-                    break;
-            }
-
-            $anonymizedIp = implode('.', $segments);
-            return $anonymizedIp;
-        } else {
-            // Not a valid IP
-            return $ip;
-        }
-    }
-
-}
+	public function getIpData($doCacheData)
+	{
+		$response = $this->fetchIpData($doCacheData);
+
+		if(!empty($response)){
+			return $response;
+		}
+	}
+
+	public function getIsEu($doCacheData)
+	{
+		$isEu = false;
+		$response = $this->fetchIpData($doCacheData);
+
+		if(!empty($response)){
+			$isEu = $response['is_eu'];
+		}
+
+		return $isEu;
+	}
+	
+	private function fetchIpData($doCacheData)
+	{
+		$devMode = craft()->config->get('devMode');
+		$ipApiKey = craft()->config->get('ipApiKey', 'geo');
+		$ip = craft()->request->getIpAddress();
+		$isEuUserCookie = 'craft_geo_eu_user';
+		$data = array(
+			'ip'=>'',
+			'country_code'=>'',
+			'is_eu'=>'',
+			'cached'=>false
+		);
+		
+		if($devMode){
+			$ip = craft()->config->get('defaultIp', 'geo');
+		}
+		
+		$cachedData = craft()->cache->get('craft.geo.' . $ip);
+
+		if ($cachedData){
+			$cached = json_decode($cachedData,true);
+			$cached['cached'] = true;
+			return $cached;
+		}
+
+		$client = new \Guzzle\Http\Client('https://api.ipstack.com/');
+		$url = $ip . '?access_key=' . $ipApiKey . '&fields=location.is_eu,country_code,ip';
+		$response = $client->get($url, array(), array('exceptions' => false))->send();
+		$data = json_decode($response->getBody());
+		return $data;
+		
+		if (!$response->getStatusCode()) {
+			return array();
+		}
+		
+		$data = array(
+			'ip'=>$data->ip,
+			'country_code'=>$data->country_code,
+			'is_eu'=>$data->location->is_eu,
+			'cached'=>true
+		);
+		
+		$isEu = $data['is_eu'] === true ? 'true' : 'false';
```

Comment:
i think we want to save $data in to the cookie and then parse EU out from the cookie
it's a little weird to have this "isEu" cookie in a function that returns an array of data.